common/src/main/java/com/microsoft/identity/common/internal/cache/AbstractAccountCredentialCache.java

```diff
@@ -176,7 +180,7 @@
                 matches = matches && equalsIgnoreCaseTrim(credentialType.name(), credential.getCredentialType());
             }
 
-            if (mustMatchOnClientId) {
+            if (mustMatchOnClientId && !(credential instanceof PrimaryRefreshTokenRecord)) {
```

Comment:
Could we instead move this logic 'up' such that its more like

```java
final boolean mustMatchOnClientId = !StringExtensions.isNullOrBlank(clientId) && !(credential instanceof PrimaryRefreshTokenRecord) ;
```

Reply:
Thanks, updated.